v-next/hardhat/test/internal/builtin-plugins/network-manager/hook-handlers/config.ts

```diff
@@ -604,6 +604,399 @@ describe("network-manager/hook-handlers/config", () => {
+          const accountsValuesToTest = [
+            { mnemonic: 123 },
+            { initialIndex: "asd" },
+            { count: "asd" },
+            { path: 123 },
+            { type: 123 },
+            {
+              initialIndex: 1,
+            },
+          ];
+
+          for (const accounts of accountsValuesToTest) {
+            hardhatUserConfig.networks.localhost.accounts = accounts;
+
+            const validationErrors =
+              await validateUserConfig(hardhatUserConfig);
+
+            assert.notEqual(validationErrors.length, 0);
+            assert.equal(
+              validationErrors[0].message,
+              "Expected 'remote', an array of strings, or an object with optional mnemonic and account details",
+            );
+          }
+        });
+      });
+
+      describe("edr config", async () => {
+        let hardhatUserConfig: any; // Use any to allow assigning also wrong values
+
+        before(() => {
+          hardhatUserConfig = {
+            networks: {
+              localhost: {
+                chainId: 1,
+                gas: "auto",
+                gasMultiplier: 1,
+                gasPrice: "auto",
+                type: "edr",
+                accounts: "", // Modified in the tests
+                url: "http://localhost:8545",
+              },
+            },
+          };
+        });
+
+        describe("allowed values", () => {
+          it("should allow an array of account objects with valid private keys", async () => {
+            hardhatUserConfig.networks.localhost.accounts = [
+              {
+                balance: "123",
+                privateKey:
+                  "0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+              },
+              {
+                balance: "123",
+                privateKey:
+                  "0xbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb",
+              },
+              {
+                balance: "123",
+                privateKey:
+                  "0xcccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc",
+              },
+            ];
+
+            const validationErrors =
+              await validateUserConfig(hardhatUserConfig);
+
+            console.log(validationErrors);
+
+            assert.equal(validationErrors.length, 0);
+          });
+
+          it("should allow an account with a valid EdrNetworkHDAccountsConfig", async () => {
+            hardhatUserConfig.networks.localhost.accounts = {
+              mnemonic: "asd asd asd",
+              initialIndex: 0,
+              count: 123,
+              path: "m/1/2",
+              accountsBalance: "123",
+              passphrase: "passphrase",
+            };
+
+            const validationErrors =
+              await validateUserConfig(hardhatUserConfig);
+
+            assert.equal(validationErrors.length, 0);
+          });
+
+          // TODO: still valid?
```

Comment:
Like the comment above:

Do we still wanna keep the possibility of allowing valid private keys with missing hex prefix?
If yes, I will have to modify the hh-utils method that I added to check if an address is valid, [see here](https://github.com/NomicFoundation/hardhat/pull/5817/files#diff-8d9a1ba6601820ff6249438ad3797d59980861eef0ca459e2a05512362894bcd)